### DISCUSSION TITLE
OQS OpenSSL build error issue on Ubuntu 22.04

### DISCUSSION THREAD

**Discussion by jingyosong** [open-quantum-safe/openssl#435] [Q&A] [answered]
hello. I am a researcher applying post quantum cryptography to protocols.

I have a question because an error occurred while building OQS OpenSSL on Ubuntu 22.04.

I first installed liboqs as described in the build guide for OQS OpenSSL.
After that, I tried to install OQS OpenSSL with the command (make -j), but I keep getting the error as below.

error: fatal error: oqs/oqs.h: No such file or directory...

![에러](https://user-images.githubusercontent.com/60771953/228128837-e4cbd25d-bbde-46f9-8c31-6a52ed1dc9f3.png)

Is there any way to solve this? Thanks for reading.

**Comment by baentsch** [comment]
This typically means that the installation of `liboqs` didn't wind up in the right location (of oqs-openssl111).

Before diving into this, how long do you want to use this software? openssl111 is nearing its end of life (see https://www.openssl.org/blog/blog/2023/03/28/1.1.1-EOL/) which also means that we will cease to support oqs-openssl111. 

Therefore, I'd like to encourage you very much to look at using [oqs-provider](https://github.com/open-quantum-safe/oqs-provider) running as a plugin in OpenSSL3 instead. You can use it ready-built e.g., in this docker image: https://hub.docker.com/repository/docker/openquantumsafe/oqs-ossl3/general

**Reply by jingyosong** [reply, marked as answer]
Thank you for answer. 

I downloaded OQS OPENSSL first and installed liboqs in its subdirectory. Can you point me to the correct location to install liboqs into oqs openssl directory?
